Lib/idlelib/idle_test/test_editor.py

```diff
@@ -1,14 +1,432 @@
+""" Test idlelib.editor.
+"""
+
 import unittest
-from idlelib.editor import EditorWindow
+import tkinter as tk
+import sys
+from functools import partial
+from idlelib import editor
+from idlelib.multicall import MultiCallCreator
+from test.support import requires
+from unittest import mock
+
+root = None
+editwin = None
+
+
+def setUpModule():
+    global root, editwin
+    requires('gui')
```

Comment:
Isn't it overly restrictive to require a GUI for all tests here? As an extreme example, testing `prepstr()` certainly doesn't require a GUI.